TheJumpCloud/support
# Sa 3606 update commands json ci

## Issues
* [SA-3606](https://jumpcloud.atlassian.net/browse/SA-3606) - Commands Gallery CI Workflow

## What does this solve?
Moving cicleci workflows and python scripts to GHActions. Added tests for functions and changes in commands gallery .json file
## Is there anything particularly tricky?

## How should this be tested?
1. Create a test branch with this branch changes
2. Edit one of the commands
3. On your terminal go to `/support/ .github/scripts`
4. Run: `python3 build_commands_gallery.py`
5. Validated changes to commands_gallery.json 
6. Now go to: `/support/ .github/scripts/tests` and run `pytest` and an error like this should pop up: 
![image](https://github.com/TheJumpCloud/support/assets/97972790/fd38421a-7095-4eae-a7c8-10d674742c35)
7. Make changes to one of the python functions that will result to an error
8. Rerun pytest again, validate that the error is caught.
9. Create a PR to Master and label it `Commands Gallery`
10. Validate in GHAction is running

## Screenshots



[SA-3606]: https://jumpcloud.atlassian.net/browse/SA-3606?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
